internal/config/config_test.go

```diff
@@ -0,0 +1,836 @@
+		assert.NoError(t, err)
+
+		expectedOutput := `{
+    "install_id": "3998b053-dd7f-4bfe-bb10-c4f3a96a0180",
+    "default_tenant": "auth0-cli.eu.auth0.com",
+    "tenants": {
+        "auth0-cli.eu.auth0.com": {
+            "name": "auth0-cli",
+            "domain": "auth0-cli.eu.auth0.com",
+            "access_token": "eyfSaswe",
+            "expires_at": "2023-04-18T11:18:07.998809Z",
+            "client_id": "secret"
+        },
+        "auth0-mega-cli.eu.auth0.com": {
+            "name": "auth0-mega-cli",
+            "domain": "auth0-mega-cli.eu.auth0.com",
+            "access_token": "eyfSaswe",
+            "expires_at": "2023-04-18T11:18:07.998809Z",
+            "client_id": "secret"
+        }
+    }
+}`
+
+		assertConfigFileMatches(t, config.path, expectedOutput)
+	})
+}
+
+func TestConfig_RemoveTenant(t *testing.T) {
+	var testCases = []struct {
+		name           string
+		givenConfig    string
+		givenTenant    string
+		expectedConfig string
+	}{
+		{
+			name:        "it can successfully remove a tenant from the config",
+			givenTenant: "auth0-mega-cli.eu.auth0.com",
+			givenConfig: createTempConfigFile(t, []byte(`{
+				"install_id": "3998b053-dd7f-4bfe-bb10-c4f3a96a0180",
+				"default_tenant": "auth0-cli.eu.auth0.com",
+				"tenants": {
+					"auth0-cli.eu.auth0.com": {
+						"name": "auth0-cli",
+						"domain": "auth0-cli.eu.auth0.com",
+						"access_token": "eyfSaswe",
+						"expires_at": "2023-04-18T11:18:07.998809Z",
+						"client_id": "secret"
+					},
+					"auth0-mega-cli.eu.auth0.com": {
+						"name": "auth0-mega-cli",
+						"domain": "auth0-mega-cli.eu.auth0.com",
+						"access_token": "eyfSaswe",
+						"expires_at": "2023-04-18T11:18:07.998809Z",
+						"client_id": "secret"
+					}
+				}
+			}`)),
+			expectedConfig: `{
+    "install_id": "3998b053-dd7f-4bfe-bb10-c4f3a96a0180",
+    "default_tenant": "auth0-cli.eu.auth0.com",
+    "tenants": {
+        "auth0-cli.eu.auth0.com": {
+            "name": "auth0-cli",
+            "domain": "auth0-cli.eu.auth0.com",
+            "access_token": "eyfSaswe",
+            "expires_at": "2023-04-18T11:18:07.998809Z",
+            "client_id": "secret"
+        }
+    }
+}`,
+		},
+		{
+			name:        "it can successfully remove the default tenant from the config",
+			givenTenant: "auth0-cli.eu.auth0.com",
+			givenConfig: createTempConfigFile(t, []byte(`{
+				"install_id": "3998b053-dd7f-4bfe-bb10-c4f3a96a0180",
+				"default_tenant": "auth0-cli.eu.auth0.com",
+				"tenants": {
+					"auth0-cli.eu.auth0.com": {
+						"name": "auth0-cli",
+						"domain": "auth0-cli.eu.auth0.com",
+						"access_token": "eyfSaswe",
+						"expires_at": "2023-04-18T11:18:07.998809Z",
+						"client_id": "secret"
+					},
+					"auth0-mega-cli.eu.auth0.com": {
+						"name": "auth0-mega-cli",
+						"domain": "auth0-mega-cli.eu.auth0.com",
+						"access_token": "eyfSaswe",
+						"expires_at": "2023-04-18T11:18:07.998809Z",
+						"client_id": "secret"
+					}
```

Comment:
Would be nice to add a third tenant here to make it clear what happens if there are multiple "fallbacks".

Reply:
In Go, maps are unordered collections of key/value pairs. When you range over a map, the order in which the key/value pairs are returned is not guaranteed and can vary between different runs of the program. We can add logic to have a consistent sorting and always set the default tenant to a more predictable one but I'd avoid it IMO as it feels unnecessary. Wdyt?

Reference: https://go.dev/blog/maps Iteration order